### PR TITLE
KAA-260: Added cassandra dao implementation.

### PR DESCRIPTION
**Pull request by shpigor**

### CODE REVIEW

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/SqlDao.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.kaaproject.kaa.server.common.dao.impl;
```
Comment (ghost) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/SqlDao.java:17]: Please, add a license header.

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/SqlDao.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public interface SqlDao<T> extends Dao<T, String> {
+
+    /**
+     * Find object by id.
+     *
+     * @param id   the id
+     * @param lazy specifies whether return initialized object (if false is set)
+     *             or proxy (if true is set)
+     * @return the found object
```
Comment (ghost) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/SqlDao.java]: Could this method return null?

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/sql/HibernateNotificationSchemaDao.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package org.kaaproject.kaa.server.common.dao.impl.sql;
```
Comment (ghost) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/impl/sql/HibernateNotificationSchemaDao.java:16]: Please, add a license header.

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/model/sql/NotificationSchema.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
+package org.kaaproject.kaa.server.common.dao.model.sql;
```
Comment (dyosick) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/model/sql/NotificationSchema.java]: Please, add a license header.

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java [hunk 5]
```diff
@@ -280,15 +278,17 @@ public boolean checkAccessToken(ApplicationDto appDto, String userExternalId, St
             return false;
         }else{
```
Comment (ghost) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java]: Please, add spaces before and after the 'else' statement.

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java [hunk 6]
```diff
@@ -297,24 +297,27 @@ public EndpointProfileDto attachEndpointToUser(String userExternalId, String ten
...
         return saveEndpointProfile(profile);
     }
 
     @Override
     public EndpointProfileDto attachEndpointToUser(String endpointUserId, String endpointAccessToken) {
+        LOG.info("Try to attach endpoint with access token {} to user with {}", endpointAccessToken, endpointUserId);
         validateString(endpointUserId, "Incorrect endpointUserId " + endpointUserId);
         EndpointUser endpointUser = endpointUserDao.findById(endpointUserId);
+        LOG.trace("[{}] Found endpoint user with id {} ", endpointUserId, endpointUser);
         if(endpointUser != null){
```
Comment (ghost) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java]: Please, add the codestyle spaces all around the source file.

File: server/common/dao/src/test/java/org/kaaproject/kaa/server/common/dao/impl/sql/HibernateNotificationSchemaDaoTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
+package org.kaaproject.kaa.server.common.dao.impl.sql;
```
Comment (ghost) [inline server/common/dao/src/test/java/org/kaaproject/kaa/server/common/dao/impl/sql/HibernateNotificationSchemaDaoTest.java:16]: License header.

File: server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/AbstractCassandraDao.java [hunk 8]
```diff
@@ -0,0 +1,193 @@
+package org.kaaproject.kaa.server.common.nosql.cassandra.dao;
```
Comment (ghost) [inline server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/AbstractCassandraDao.java:17]: License header.

File: server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+        LOG.debug("Endpoint profile saved");
+        return profile;
+    }
+
+    @Override
+    public CassandraEndpointProfile findByKeyHash(byte[] endpointKeyHash) {
+        LOG.debug("Try to find endpoint profile by key hash [{}]", endpointKeyHash);
+        CassandraEndpointProfile endpointProfile = (CassandraEndpointProfile) getMapper().get(getByteBuffer(endpointKeyHash));
+        LOG.debug("{} endpoint profile by key hash [{}]", endpointKeyHash, endpointProfile != null ? "Found" : "No found");
+        LOG.trace("Found endpoint profile {} by key hash [{}]", endpointKeyHash, endpointProfile);
```
Comment (ghost) [inline server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java]: I think these two log traces are contradictory.

File: server/common/nosql/mongo-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/mongo/dao/AbstractMongoDao.java [hunk 10]
```diff
@@ -107,50 +102,37 @@ protected long count(Query query) {
...
-    public T findById(String id) {
-        return findById(id, false);
-    }
-
-    @Override
-    public T findById(String id, boolean lazy) {
-        LOG.debug("Find document of collection [{}] by id [{}]", getCollectionName(), id);
-        return mongoTemplate.findById(id, getDocumentClass());
+    public T findById(K key) {
+        LOG.debug("Find document of collection [{}] by id [{}]", getCollectionName(), key);
```
Comment (ghost) [inline server/common/nosql/mongo-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/mongo/dao/AbstractMongoDao.java:122]: I propose to use one of a log pattern, either "Find..." or "Try to find...", but not both.

File: pom.xml [hunk 11]
```diff
@@ -107,7 +109,7 @@ Copyright 2014 CyberVision, Inc.
                 <activeByDefault>true</activeByDefault>
             </activation>
             <properties>
-                <build.commit>N/A</build.commit>
+                <build.commit>${GIT_COMMIT}</build.commit>
```
Comment (ashvayka) [inline pom.xml]: Please remove this property if it is not used anymore

File: server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointNotificationCassandraDao.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @Override
+    public void removeById(String id) {
+        LOG.debug("Remove endpoint notifications by id {}", id);
+        CassandraEndpointNotification key = new CassandraEndpointNotification(id);
+        Statement delete = delete().from(getColumnFamilyName()).where(eq(ET_NF_ENDPOINT_KEY_HASH_PROPERTY, key.getEndpointKeyHash()))
+                .and(eq(ET_NF_LAST_MOD_TIME_PROPERTY, key.getLastModifyTime()));
+        execute(delete);
+        LOG.debug("[id] Execute query {}:", id, delete);
```
Comment (ashvayka) [inline server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointNotificationCassandraDao.java]: Maybe this should be LOG.debug("[{}] Execute query: {}", id, delete); ?

File: server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+        if (accessToken != null) {
+            saveByAccessToken = cassandraEPByAccessTokenDao.getSaveQuery(new CassandraEPByAccessToken(accessToken, epKeyHash));
+        }
+        Statement saveProfile = getSaveQuery(profile);
+        if (saveByAccessToken != null) {
+            executeBatch(saveProfile, saveByAppId, saveByAccessToken);
+        } else {
+            executeBatch(saveProfile, saveByAppId);
+        }
+        LOG.debug("Endpoint profile saved");
```
Comment (ashvayka) [inline server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java]: add id of the endpoint profile here.

**Comment by ashvayka** [comment]
Some minor comments about logging messages - please review all log messages. Otherwise LGTM.

**Pull request merged**